src/main/java/com/google/devtools/build/lib/rules/java/JavaToolchainTool.java

```diff
@@ -105,7 +105,7 @@ private CustomCommandLine extractCommandLine(JavaToolchainProvider toolchain)
 
     Artifact executable = tool().getExecutable();
     if (!executable.getExtension().equals("jar")) {
-      command.addExecPath(executable);
+      command.addExecPath(executable).addAll(jvmOpts());
```

Comment:
Hyrums law and the comment on L67 has me thinking this is a little risky. We can hope this is never set when the tools is not a jar, but we may end up having to add a separate parameter. But lets see.

Reply:
As far as I understand the code, `jvmOpts` cannot be set to a non-empty value for the direct header compiler via `java_toolchain`, so I'm hopeful that this change won't cause issues.